datastore/key.go

```diff
@@ -105,19 +105,29 @@ func (k *Key) marshal(b *bytes.Buffer) {
 	} else {
 		b.WriteString(strconv.FormatInt(k.ID, 10))
 	}
-	if k.Namespace != "" {
+	if k.Namespace != "" && includeNamespace {
 		b.WriteByte(',')
 		b.WriteString(k.Namespace)
 	}
 }
 
-// String returns a string representation of the key.
+// String returns a string representation of the key. It does not include
+// the namespace in case that the Namespace's name is sensitive information.
 func (k *Key) String() string {
+	return k._string(false)
+}
+
+// StringWithNamespace is identical to String(), but appends the namespace.
+func (k *Key) StringWithNamespace() string {
+	return k._string(true)
+}
+
+func (k *Key) _string(includeNamespace bool) string {
```

Comment:
Yeah, I nearly didn't do this, but redefining `string` as a func identifier when it clashes with a builtin type felt bad, too.  I guess since it takes a receiver it's fine to call it `Key.string`, though. Will change.